ssl/src/main/java/io/crate/protocols/ssl/SslHandlerLoader.java

```diff
@@ -20,38 +20,63 @@
  * agreement.
  */
 
-package io.crate.protocols.postgres.ssl;
+package io.crate.protocols.ssl;
 
+import io.crate.protocols.http.HttpsHandler;
+import io.crate.protocols.http.DefaultHttpsHandler;
+import io.crate.protocols.postgres.SslReqHandler;
+import io.crate.protocols.postgres.SslReqRejectingHandler;
+import io.crate.settings.CrateSetting;
 import io.crate.settings.SharedSettings;
 import org.apache.logging.log4j.Logger;
 import org.elasticsearch.common.logging.Loggers;
 import org.elasticsearch.common.settings.Settings;
 
 import java.lang.reflect.InvocationTargetException;
+import java.util.function.Supplier;
 
 /**
- * Loads the appropriate implementation of the SslReqHandler.
+ * Loads the appropriate implementation of the PSQL SSL and Https handlers.
  */
-public class SslReqHandlerLoader {
+public class SslHandlerLoader {
 
-    private static final Logger LOGGER = Loggers.getLogger(SslReqHandlerLoader.class);
-    private static final String SSL_IMPL_CLASS = "io.crate.protocols.postgres.ssl.SslReqConfiguringHandler";
+    private static final Logger LOGGER = Loggers.getLogger(SslHandlerLoader.class);
 
-    private SslReqHandlerLoader() {}
+    private SslHandlerLoader() {}
 
     /**
      * Loads the SslRequest handler. Should only be called once per application life time.
      * @throws SslConfigurationException Exception thrown if the user has supplied an invalid configuration
      */
-    public static SslReqHandler load(Settings settings) {
+    public static SslReqHandler loadSslReqHandler(Settings settings) {
+        return load(settings,
+            SslConfigSettings.SSL_PSQL_ENABLED,
+            "io.crate.protocols.postgres.SslReqConfiguringHandler",
```

Comment:
I was wondering whether this should be exposed at all. Since there are multiple messages now, I wanted to ensure the classes don't get mixed up. However, there are type checks in place, so should be safe.